File: packages/people/ios/Sources/PeoplePlugin/models/ContactModels.swift
import Foundation
import Contacts

/// Nominal model for organization data to ensure cross-platform parity.
public struct ContactOrganization {
    public let company: String?
    public let title: String?
    public let department: String?
}

/// Nominal model for a phone field.
public struct ContactPhone {
    public let label: String?
    public let number: String
}

/// Nominal model for an email field.
public struct ContactEmail {
    public let label: String?
    public let address: String
}

/// Nominal model for a postal address field.
public struct ContactAddress {
    public let label: String?
    public let formatted: String?
    public let street: String?
    public let city: String?
    public let region: String?
    public let postcode: String?
    public let country: String?
}

/// Nominal model representing contact data, independent of Capacitor.
/// This ensures type safety within the Implementation layer.
public struct ContactData {
    public let id: String
    public let displayName: String?
    public let firstName: String?
    public let lastName: String?
    public let organization: ContactOrganization? // Updated from String? to struct
    public let birthday: DateComponents?
    public let phones: [ContactPhone]?
    public let emails: [ContactEmail]?
    public let addresses: [ContactAddress]?
    public let urls: [String]?
    public let note: String?
    public let image: String? // Base64 string
}

/// Nominal model for Group data to replace raw dictionaries.
public struct GroupData {
    public let id: String
    public let name: String
    public let source: String
    public let readOnly: Bool
}

/// Result model for paginated contact fetches.
public struct GetContactsResultData {
    public let contacts: [ContactData]
    public let totalCount: Int
}

/// Result model for group listings.
public struct ListGroupsResultData {
    public let groups: [GroupData]
}

File: packages/people/android/src/main/java/io/capkit/people/models/ContactModels.kt
package io.capkit.people.models

/**
 * Native representation of a Contact.
 *
 * Architectural rules:
 * - This class MUST remain agnostic to Capacitor JSObjects.
 * - Used exclusively by the Impl layer to orchestrate native data.
 * - Mapped to JS payloads only within the Utility or Plugin layer.
 */
data class ContactData(
  val id: String,
  val lookupKey: String? = null,
  val displayName: String? = null,
  val organization: ContactOrganization? = null, // Added for parity
  val phones: List<ContactField> = emptyList(),
  val emails: List<ContactField> = emptyList(),
  val addresses: List<PostalAddress> = emptyList(), // Extraction logic needed
)

/**
 * Native representation of a contact's organization.
 */
data class ContactOrganization(
  val company: String?,
  val title: String?,
  val department: String?,
)

// -----------------------------------------------------------------------------
// Support Models
// -----------------------------------------------------------------------------

/**
 * Represents a generic contact field (e.g., Phone or Email).
 * Includes a label (type) and the actual value.
 */
data class ContactField(
  val label: String?,
  val value: String?,
)

/**
 * Native representation of a physical postal address.
 * Maps to structured postal data from the Android Contacts provider.
 */
data class PostalAddress(
  val label: String?,
  val street: String?,
  val city: String?,
  val region: String?,
  val postcode: String?,
  val country: String?,
)

/**
 * Native representation of a contact group.
 */
data class GroupData(
  val id: String,
  val name: String,
  val source: String,
  val readOnly: Boolean,
)
